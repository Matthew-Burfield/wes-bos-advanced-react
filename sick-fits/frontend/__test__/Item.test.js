import Item from "../components/Item";
import { fakeItem } from "../lib/testUtils";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

describe("<Item />", () => {
  const fakeItem = {
    title: "Test item",
    price: 5000,
    description: "This is the description of the fake item",
    image: "image.jpg",
    largeImage: "image-lg.jpg"
  };
  const wrapper = shallow(<Item item={fakeItem} />);

  it("should render", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should render the title", () => {
    const Title = wrapper.find("Title");
    expect(Title).toBeDefined();
    expect(Title.find("a").text()).toEqual(fakeItem.title);
  });

  it("should render the PriceTag", () => {
    const PriceTag = wrapper.find("PriceTag");
    expect(PriceTag).toBeDefined();
    expect(PriceTag.dive().text()).toEqual("$50");
  });

  it("should render the image", () => {
    const img = wrapper.find("img");
    expect(img.props().src).toEqual(fakeItem.image);
    expect(img.props().alt).toEqual(fakeItem.title);
  });

  it("should render the AddToCart", () => {
    const AddToCart = wrapper.find("AddToCart");
    expect(AddToCart).toBeDefined();
  });

  it("should render the DeleteItem", () => {
    const DeleteItem = wrapper.find("DeleteItem");
    expect(DeleteItem).toBeDefined();
  });
});
