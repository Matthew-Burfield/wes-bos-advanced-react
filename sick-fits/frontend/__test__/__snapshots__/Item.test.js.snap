// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> should render 1`] = `
<ItemStyles__Item>
  <img
    alt="Test item"
    src="image.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": undefined,
          },
        }
      }
    >
      <a>
        Test item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50
  </PriceTag>
  <p>
    This is the description of the fake item
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": undefined,
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddToCart
      item={
        Object {
          "description": "This is the description of the fake item",
          "image": "image.jpg",
          "largeImage": "image-lg.jpg",
          "price": 5000,
          "title": "Test item",
        }
      }
    />
    <DeleteItem>
      Delete this item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
